lib/pbench/server/unpack_tarballs.py

```diff
@@ -0,0 +1,59 @@
+from logging import Logger
+import os
+from pathlib import Path
+import glob
+
+from pbench.server import PbenchServerConfig
+from pbench.server.filetree import FileTree
+from pbench.server.utils import get_tarball_md5
+
+
+class Unpack:
+    def __init__(self, config: PbenchServerConfig, linksrc: str, logger: Logger):
+        """Unpacks Tarball in the INCOMING Directory
+
+        Args:
+            config: PbenchServerConfig configuration object
+            logger: A Pbench python Logger
+            linksrc: "TO-UNPACK" state directory
+        """
+        self.config = config
+        self.linksrc = linksrc
+        self.logger = logger
+
+    def unpack_tarball(self) -> (int, int):
+        """Runs through a list of tarball symlinks and unpacks them using
+        FileTree's unpack module.
+
+        Returns:
+            Tuple with Total and Successful tarballs.
+        """
+        tarlist = glob.iglob(
+            os.path.join(self.config.ARCHIVE, "*", self.linksrc, "*.tar.xz")
+        )
+        ntotal = nsuccess = 0
+
+        file_tree = FileTree(self.config, self.logger)
+
+        for tarball in sorted(tarlist):
+            ntotal += 1
+
+            try:
+                tb = Path(tarball).resolve(strict=True)
+            except FileNotFoundError:
+                self.logger.error(
+                    "Tarball link, '{}', does not resolve to a real location", tarball
+                )
+                continue
+
+            tar_md5 = get_tarball_md5(tb)
```

Comment:
Given that you're trying to scan a whole directory (or several...) and report the number of successes, we should probably have the call to `get_tarball_md5()` in a `try` block, too.